jupyterhub/handlers/base.py

```diff
@@ -336,7 +336,11 @@ def login_user(self, data=None):
         if authenticated:
             username = authenticated['name']
             auth_state = authenticated.get('auth_state')
+            admin = authenticated.get('admin')
             user = self.user_from_username(username)
+            # Only set `admin` if the authenticator returned an explicit value.
+            if admin is not None:
+                user.admin = admin
```

Comment:
We need to make sure that any changes are committed before any `yield`, otherwise bad things can happen. Let's check if this is a change (`admin is not None and admin != user.admin`) and then add a `db.commit()` after assigning the new value.

Reply:
Oops, I was looking at the commit [on line 347](https://github.com/jupyterhub/jupyterhub/blob/bd36962643fa559e4ad661712c253e9e7db18fd0/jupyterhub/handlers/base.py#L347) and thought that was sufficient.

Incidentally, that `commit()` call doesn't do anything, since both `self.user_from_username` and `user.save_auth_state` call a `commit()` themselves...